repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/attributemappingvalidation/AttributeMappingPolicyValidationFilterTest.groovy

```diff
@@ -69,168 +71,119 @@ class AttributeMappingPolicyValidationFilterTest extends ReposeValveTest {
-                  }
-                ],
-                "version":"RAX-1"
-              }
-            }
-            """
+            |---
+            |mapping:
+            |  rules:
+            |  - local:
+            |      user:
+            |        name: "{D}"
+            |    remote:
+            |    - multiValue: false
+            |      name: Username
+            |      regex: false
+            |  version: RAX-1
+            """.stripMargin()
 
         when:
-        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "PUT", headers: ["content-type": APPLICATION_JSON], requestBody: body)
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "PUT", headers: ["content-type": TEXT_YAML], requestBody: body)
 
         then:
         mc.receivedResponse.code == "200"
         mc.handlings.size() == 1
-        jsonSlurper.parseText(body) == jsonSlurper.parseText(mc.handlings[0].request.body as String)
+        yaml.load(body) == yaml.load(mc.handlings[0].request.body as String)
     }
 
-    def "should not validate bad XML"() {
+    def "should fail to validate bad YAML"() {
         given:
         String body =
-            """<?xml version="1.0" encoding="UTF-8"?>
-                <mapping xmlns="http://docs.rackspace.com/identity/api/ext/MappingRules"
-                         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-                         xmlns:xs="http://www.w3.org/2001/XMLSchema"
-                         xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion"
-                         version="RAX-1">
-                   <rules>
-                      <rule>
-                        <local>
-                            <user>
-                               <name value="{D}"/>
-                               <email value="{D}"/>
-                               <expire value="{D}"/>
-                               <domain value="{D}"/>
-                               <roles value="{D}"/>
-                            </user>
-                         </local>
-                      </rule>
-                   </rules>
             """
+            |---
+            |mapping:
+            |  rules:
+            |    local:
+            |      user:
+            |        domain: "{D}"
+            |        name: "{D}"
+            |        email: "{D}"
+            |        roles: "{D}"
+            |        expire: "{D}"
+            |  version: RAX-1
+            """.stripMargin()
 
         when:
-        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "PUT", headers: ["content-type": APPLICATION_XML], requestBody: body)
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: "PUT", headers: ["content-type": TEXT_YAML], requestBody: body)
 
         then:
         mc.receivedResponse.code == "400"
         mc.handlings.size() == 0
     }
 
-    def "should not validate bad JSON"() {
+    def "YAML comments should be preserved"() {
```

Comment:
Yeah, that made less sense for your work, though, since it just fetches the policy. Totally fits here, though.